File: Backend/Nonagon.Modular.DynamicForm/FormLayoutWidth.cs
using System;

namespace Nonagon.Modular.DynamicForm
{
	/// <summary>
	/// Form layout size in horizontal.
	/// </summary>
	public enum FormLayoutWidth
	{
		/// <summary>
		/// The size is growing automatically by the children.
		/// </summary>
		WrapForm,
		
		/// <summary>
		/// The size adjusted to be the parent size.
		/// </summary>
		FillParent,
		
		/// <summary>
		/// The size adjusted to be average with their sibling children.
		/// </summary>
		Average
	}
}

File: Frontend/GWT/gwt-modular-dynamicform/src/com/nonagon/modular/dynamicform/client/FormLayoutWidth.java
package com.nonagon.modular.dynamicform.client;

public enum FormLayoutWidth {

	WrapContent,
	FillParent,
	Average;
	
	public static FormLayoutWidth fromString(String value) {

		return Enum.valueOf(FormLayoutWidth.class, value);
	}
	
	public static String fromEnum(FormLayoutWidth formLayoutWidth) {
		
		return formLayoutWidth.toString();
	}
}
